hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/read/S3File.java

```diff
@@ -98,6 +106,7 @@ public S3File(
     this.streamStatistics = streamStatistics;
     this.changeTracker = changeTracker;
     this.s3Objects = new IdentityHashMap<InputStream, S3Object>();
+    this.futurePool = context.getFuturePool();
```

Comment:
if this is only for the drain, given the context is already stored, you can just get the pool when needed